astropy/modeling/functional_models.py

```diff
@@ -440,6 +442,41 @@ def inverse(self):
         return inv
 
 
+class RedshiftSpectrum(Model):
+    """
+    One dimensional redshift model.
+
+    Parameters
+    ----------
+    z : int or float
+        Redshift value of the output spectrum.
+
+    z_in : int or float
```

Comment:
This is interesting to allow two `z` values, but is it necessary? Usually people shift from 0 to `z`, or `z` to 0. I don't have a preference, but I am just wondering if this extra complexity is necessary.